ui/src/app/pages/data/data-set/genomic-extraction-modal.tsx

```diff
@@ -79,16 +80,20 @@ export const GenomicExtractionModal = ({
         </span>
       </TextColumn>
     </ModalBody>
-    {runningExtract &&
-      <WarningMessage iconSize={30} iconPosition={'top'} data-test-id='running-extract-warning'>
-        An extraction is currently running for this dataset; it was started {TimeAgoWithVerboseTooltip(runningExtract.submissionDate)}
-      </WarningMessage>
-    }
-    {!runningExtract && succeededExtract &&
-      <WarningMessage iconSize={30} iconPosition={'top'} data-test-id='preexisting-extract-warning'>
+    {(runningExtract || succeededExtract || failedExtract) &&
```

Comment:
I think a slightly easier way to express this would be just `mostRecentExtract` and then you can have a `switchCase` inside with the `mostRecentExtract.status` as the argument.

Reply:
`mostRecentExtract` might have status `ABORTING` or `ABORTED`, though. in which case I wouldn't want to display the warning at all.